overlord/devicestate/handlers_install.go

```diff
@@ -111,6 +112,24 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 		return fmt.Errorf("cannot make run system bootable: %v", err)
 	}
 
+	// support dropping cloud-init for grade: dangerous
+	cloudCfg := filepath.Join(dirs.RunMnt, "ubuntu-seed/cloud.cfg.d")
+	if osutil.IsDirectory(cloudCfg) && deviceCtx.Model().Grade() == asserts.ModelDangerous {
```

Comment:
Would it be worth logging that there was cloud.cfg.d in the seed, but that it was ignored due to grade != dangerous.  It might be a nice breadcrumb for users trying to get cloud-config into an device and don't understand why their cloud-config isn't present in the device after an install.